src/index.ts

```diff
@@ -1688,12 +1701,14 @@ export class CachedMerklePatriciaTree<K, V> extends MerklePatriciaTree<K, V> {
    *
    * @returns value corresponding to the key if present; null if otherwise
    */
-  getFromCache(key: K, nodeMap: Map<bigint, MerklePatriciaTreeNode<V>>): V {
+  getFromCache(
+      key: K, nodeMap: Map<bigint, MerklePatriciaTreeNode<V>>,
+      cache: Array<bigint> = []): CachedTraversal<V> {
```

Comment:
not sure why I would want this as a parameter?

In this issue request I suggest an array of nodeMaps as a vararg parameter

Reply:
If you are maintaining a cache of most recently used nodes; you can send it in the request; and the getFromCache does the following:

If the node is not present in the cache, adds it to the cache
If it is present in the cache; it moves the node to the end of the cache.

So; if you are going to maintain one array<bigint> cache; most recent accessed nodes will be towards the end of the cache. Does this make sense?

Reply:
I don't understand what this array is for. It doesn't even contain the nodes, so it doesn't contain any data, so it most certainly isn't a cache. Maybe it's a "list" of nodes used, which is automatically updated by the Merkle tree.

Fine, then I guess we don't need CachedTraversal<V> as the return type, since it is redundant.

I suggest this. Rename the parameter to `bagNodesUsed`, and make it a `Set<bigint>`.

Reply:
Oh seems like I misunderstood your requirement... now I actually understand your requirement.

I can change it as you suggested ? Or should I just rename the variable and make it a set?
@no2chem

Reply:
Just make it a set for now, and change the return type back.

Reply:
Okay sure.